File: Remove_Duplicates_from_Shorted_Array.cpp
// Title: 80. Remove Duplicates from Sorted Array II
// Difficulty: Medium
// Problem: https://leetcode.com/problems/remove-duplicates-from-sorted-array-ii/description/

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() == 1)
        {
            return 1;
        }
        int k = 2;
        for (int i=2; i<nums.size(); i++)
        {
            if (nums[i] != nums[k-2])
            {
                nums[k] = nums[i];
                k++;
            }
        }
        return k;
    }
};

File: Remove_Duplicates_from_Shorted_Array.py
# Title: 80. Remove Duplicates from Sorted Array II
# Difficulty: Medium
# Problem: https://leetcode.com/problems/remove-duplicates-from-sorted-array-ii/description/

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        shift = 0
        for i in range(2,len(nums)):
            i -= shift
            if nums[i] == nums[i-1] and nums[i-1] == nums[i-2]:
                nums.pop(i)
                shift += 1
        return len(nums)

